### ISSUE TITLE
[css-position] Proposal for new 'position' value: fixed-to-parent

### ISSUE THREAD

**Issue by jez9999** [w3c/csswg-drafts#8066]
At present the CSS `position` can be set to static, relative, absolute, fixed, or sticky:
https://w3c.github.io/csswg-drafts/css-position/#position-property

I'd like a new value added: `fixed-to-parent`.  This would behave similarly to `fixed`, but the element would instead be positioned and sized relative to the nearest block-level ancestor element.

**Motivation**
I've been trying to create a "mask" element over my overflowing content, which receives a scrollbar, whereby the mask itself doesn't overlay the scrollbar.  The closest I came to sort-of getting it working was this:
https://game-point.net/misc/stickyMaskOverScroll/index.htm

It uses a combination of grid layout and sticky to get the mask to stay fixed relative to its container, allowing the sibling's content to overflow and scroll, but it doesn't work in Firefox and I think it's lucky it works in Chrome because it doesn't really seem to be guaranteed behaviour in the CSS spec; as I'm using `max-height` instead of `height` to size the container, the height of the grid's row should be set to `auto`, so really the mask element should not have the height that it does.

I think what I'm trying to do is a fundamentally new behaviour which, whilst it would seem to be relatively feasible to implement with existing rendering engines, isn't currently producible.  It surprises me that there isn't a way to do this currently as it's quite common to have content scrolling through overflow, and if you wanted to put a mask over the viewing area / container, or maybe fix some widget relative to the container, you couldn't do it right now.

**Comment by Loirooriol** [comment]
No, the problem in Firefox is [bug 1481876](https://bugzilla.mozilla.org/show_bug.cgi?id=1481876), which prevents `grid-template-rows: 100%` to resolve against the final height of the grid container. It basically behaves as `grid-template-rows: auto`. So the row (height of the containing block for the sticky) ends up being larger than the height of the scroll container.

Just use `grid-template-rows: 300px` or `grid-template-rows: minmax(0, 1fr)` and it will work.

Your `fixed-to-parent` proposal seems like `position: absolute` with `position: relative` on the parent.

**Comment by jez9999** [comment]
Wow... `grid-template-rows: minmax(0, 1fr)` actually does work.  I never considered using 'fraction' like that, probably because it's such a new measurement!  Thanks!  Yes, on careful reading of the spec, it looks like the 100% should also work and that one's a bug in Fx.

> Your `fixed-to-parent` proposal seems like `position: absolute` with `position: relative` on the parent.

Well, no, because that causes the overlaid content to scroll with the scrollable content.  This was to fix the content in place, which now seems to have been solved with grid, sticky, and crucially setting the grid row height with minmax(0,1fr) instead of percentage.

**Comment by benface** [comment]
So, maybe OP's example wasn't the best vehicle to propose this idea, but I think a new position value that behaves like `fixed` but is relative to the nearest scroll ancestor (not scrolling with it, as opposed to `absolute`) instead of the viewport is a great idea, and definitely something I've needed in the past. In fact, it's obviously too late to change that, but `fixed` would have been a great name for it, and the current `fixed` could have been called `viewport-fixed`. How about `scroll-fixed`?
